flytekit/core/array_node.py

```diff
@@ -0,0 +1,214 @@
+    Promise,
+    VoidPromise,
+    flyte_entity_call_handler,
+    translate_inputs_to_literals,
+)
+from flytekit.core.task import TaskMetadata
+from flytekit.loggers import logger
+from flytekit.models import literals as _literal_models
+from flytekit.models.core import workflow as _workflow_model
+from flytekit.models.literals import Literal, LiteralCollection, Scalar
+
+
+class ArrayNode(object):
+    def __init__(
+        self,
+        target: LaunchPlan,
+        concurrency: Optional[int] = None,
+        min_successes: Optional[int] = None,
+        min_success_ratio: Optional[float] = None,
+        bound_inputs: Optional[Set[str]] = None,
+        execution_version: Optional[int] = None,
+        metadata: Optional[Union[_workflow_model.NodeMetadata, TaskMetadata]] = None,
+    ):
+        """
+        :param target: The target Flyte entity to map over
+        :param concurrency: If specified, this limits the number of mapped tasks than can run in parallel to the given batch
+            size. If the size of the input exceeds the concurrency value, then multiple batches will be run serially until
+            all inputs are processed. If set to 0, this means unbounded concurrency. If left unspecified, this means the
+            array node will inherit parallelism from the workflow
+        :param min_success_ratio: If specified, this determines the minimum fraction of total jobs which can complete
+            successfully before terminating this task and marking it successful.
+        :param min_successes: If specified, an absolute number of the minimum number of successful completions of subtasks.
+            As soon as the criteria is met, the array job will be marked as successful and outputs will be computed.
+        """
+        self.target = target
+        self._concurrency = concurrency
+        self._min_successes = min_successes
+        self._min_success_ratio = min_success_ratio
+        self._execution_version = execution_version
+        self.id = target.name
+
+        n_outputs = len(self.target.python_interface.outputs)
+        if n_outputs > 1:
+            raise ValueError("Only tasks with a single output are supported in map tasks.")
+
+        self._bound_inputs: Set[str] = bound_inputs or set(bound_inputs) if bound_inputs else set()
+
+        output_as_list_of_optionals = min_success_ratio is not None and min_success_ratio != 1 and n_outputs == 1
+        collection_interface = transform_interface_to_list_interface(
+            self.target.python_interface, self._bound_inputs, output_as_list_of_optionals
+        )
+        self._collection_interface = collection_interface
+
+        self.metadata = None
+        if isinstance(target, LaunchPlan):
+            if self._execution_version is None:
+                self._execution_version = 1
+            if self._execution_version != 1:
+                raise ValueError("Only execution version 1 is supported for LaunchPlans.")
```

Comment:
can you clarify what's this used for?

Reply:
Also, can we make `execution_version` a required parameter?

Reply:
I think the enum is preferable. I'll update that soon.

Yup, will make execution_version required.